### PR TITLE
Cleanup klog.sh debugging script

### PR DESCRIPTION
**Pull request by alexjh**
* shellcheck linting
* Simplify klog by removing error checks. This will be handled by another tool in the future.
* Don't remove this pod suffix - this was overwriting log files if there was more than one pod.
* Get some extra kube logs related to each pod and the overall namespace

### CODE REVIEW

File: container-host-files/opt/scf/bin/klog.sh [hunk 1]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
-# etcd pod /var/vcap/sys/log/etcd/etcd.stderr.log
-# duplicate of previous error?
-lookfor <<EOF
-etcdserver: publish error: etcdserver: request timed out
+    PODS=$(kubectl get pods --namespace "${NS}" --output name --show-all=true | sed 's/pods\///')
 
-${MSG}
-EOF
+    for POD in ${PODS}; do
+        DIR=${KLOG}/${NS}/${POD}
```
Comment (mook-as) [inline container-host-files/opt/scf/bin/klog.sh]: Quoting?

File: container-host-files/opt/scf/bin/klog.sh [hunk 2]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
-EOF
+if [ ! -f "${DONE}" ]; then
+    mkdir -p "${KLOG}/${NS}"
+    rm -rf "${KLOG:?}/${NS:?}/"*
 
-# etcd pod /var/vcap/sys/log/etcd/etcd.stderr.log
-# duplicate of previous error?
-lookfor <<EOF
-etcdserver: publish error: etcdserver: request timed out
+    PODS=$(kubectl get pods --namespace "${NS}" --output name --show-all=true | sed 's/pods\///')
```
Comment (mook-as) [inline container-host-files/opt/scf/bin/klog.sh]: If we make this an array `PODS=($(kubectl …))`, we can use `"${PODS[@]}"` later for safer quoting. Not that it actually makes a difference…

File: container-host-files/opt/scf/bin/klog.sh [hunk 3]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
-4.0.1 clusters, the fix is complicated as it involves manually
-patching.
-EOF
-
-# consul pod /var/vcap/sys/log/consul_agent/consul_agent.stdout.log
-lookfor <<EOF
-Only one node should be in bootstrap mode, not adding Raft peer.
+        # Get the CF logs inside the pod if there are any
+        if [ "$(kubectl get pod "${POD}" --namespace "${NS}" --output=jsonpath='{.status.phase}')" != 'Succeeded' ] && \
+                kubectl exec "${POD}" --namespace "${NS}" -- bash -c "[ -d /var/vcap/sys/log ]" 2> /dev/null; then
```
Comment (mook-as) [inline container-host-files/opt/scf/bin/klog.sh]: These lines are somewhat giant. Can we put those in variables or something?

File: container-host-files/opt/scf/bin/klog.sh [hunk 4]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
+            # Mask the exit status of tar because it complains if files were written while it was reading them
+            kubectl exec "${POD}" --namespace "${NS}" -- bash -c "cd /var/vcap/sys/log && (tar --warning=no-file-changed -cf - * || true)" | ( cd "${DIR}" && tar xf -)
```
Comment (mook-as) [inline container-host-files/opt/scf/bin/klog.sh]: `-- tar -C /var/vcap/sys/log --warning=no-file-changed …`
`… tar -C "${DIR}" xf -`

But then, this is basically `kubectl cp --namespace "${NS}" "${POD}:/var/vcap/sys/log/" "${DIR}/"`

File: container-host-files/opt/scf/bin/klog.sh [hunk 5]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
+        fi
 
-This occurs when two consul nodes race to bootstrap. Usually deleting
-the consul pods (so HCP makes new ones) fixes this.
-EOF
+        # Get the pod logs - previous may not be there if it was successful on the first run.
+        # Unfortunately we can't get anything past the previous one
+        kubectl logs "${POD}" --namespace "${NS}" > "${DIR}/kube.log"
+        kubectl logs "${POD}" --namespace "${NS}" --previous > "${DIR}/kube-previous.log" 2> /dev/null || true
+        kubectl describe pods "${POD}" --namespace "${NS}" > "${DIR}/describe-pod"
```
Comment (mook-as) [inline container-host-files/opt/scf/bin/klog.sh]: Add a `.txt` extension?

File: container-host-files/opt/scf/bin/klog.sh [hunk 6]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
+    done
+    gunzip -r "${KLOG}"
```
Comment (mook-as) [inline container-host-files/opt/scf/bin/klog.sh]: This could use a comment about why (logrotate, presumably?)

Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: I don't know why this is here. I agree it's probably for logrotate, but I don't have a setup that's run long enough to generate rotated logs. I've left it alone other than adding quotes to make shellcheck happy.

File: container-host-files/opt/scf/bin/klog.sh (hunk 5 above)
Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: Fixed in 1a20f22d

File: container-host-files/opt/scf/bin/klog.sh [hunk 7]
```diff
@@ -22,152 +22,40 @@ if [ "$1" == "-f" ]; then
...
-
-# etcd pod /var/vcap/sys/log/etcd/etcd_ctl.err.log
-lookfor <<EOF
-Error:  cannot sync with the cluster using endpoints https://etcd-
-
-${MSG}
-EOF
+if [ ! -f "${DONE}" ]; then
+    mkdir -p "${KLOG}/${NS}"
+    rm -rf "${KLOG:?}/${NS:?}/"*
```
[the remark this answers is not in the document]
Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: Fixed in 3d6e794f

File: container-host-files/opt/scf/bin/klog.sh (hunk 3 above)
Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: Fixed in 996336af

File: container-host-files/opt/scf/bin/klog.sh (hunk 4 above)
Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: TIL about `cp` :grinning: 

Fixed in 4442684c

File: container-host-files/opt/scf/bin/klog.sh (hunk 2 above)
Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: :+1: 

Fixed in 6df1b836

File: container-host-files/opt/scf/bin/klog.sh (hunk 1 above)
Comment (alexjh) [reply container-host-files/opt/scf/bin/klog.sh]: Fixed in 008b91e5

**Pull request merged**